### PR TITLE
kdocker: init at 5.4

### LINKED ISSUE #218567: Package request: kdocker
**Project description**

KDocker will help you dock any application in the system tray. This means you can dock openoffice, firefox, thunderbolt, eclipse, anything! Just point and click. Works for both KDE and GNOME (In fact it should work for most modern window managers that support NET WM Specification. I believe it works for XFCE, for instance)

All you need to do is start KDocker and select an application using the mouse and lo! the application gets docked into the system tray. The application can also be made to dissappear from the task bar.

**Metadata**

* homepage URL: https://github.com/user-none/KDocker
* source URL: https://github.com/user-none/KDocker
* license: gpl2.0
* platforms: linux

### PR DESCRIPTION
**Pull request by hexclover**
## Description of changes

Added a new package `kdocker`, a tool for docking any application.

Resolves https://github.com/NixOS/nixpkgs/issues/218567.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/kdocker/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, qtbase, wrapQtAppsHook, fetchFromGitHub, qmake, qtx11extras, libX11, libXmu, libXpm }:
```
Comment (drupol) [inline pkgs/tools/X11/kdocker/default.nix:1]:
```suggestion
{ stdenv
, lib
, qtbase
, wrapQtAppsHook
, fetchFromGitHub
, qmake
, qtx11extras
, libX11
, libXmu
, libXpm
}:

```

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your first contribution!

It's almost perfect,  I've just left one minor comment, left me know if this is clear enough.

File: pkgs/tools/X11/kdocker/default.nix (hunk 1 above)
Comment (hexclover) [reply pkgs/tools/X11/kdocker/default.nix]: Done! Also fixed a mistake in the patching phase.

File: pkgs/tools/X11/kdocker/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    for h in Xatom Xlib Xmu; do
+      sed -i "s|#include <$h|#include <X11/$h|" src/xlibutil.h src/{kdocker,scanner,trayitem,trayitemmanager}.cpp
+    done
+    for t in target icons desktop appdata; do
+      sed -i "s|$t.path = /usr|$t.path = $out|" kdocker.pro
+    done
+    sed -i "s|/etc/bash_completion.d|$out/share/bash-completion/completions|" kdocker.pro
+  '';
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/tools/X11/kdocker/default.nix:47]: Consider adding yourself as maintainer in here.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/tools/X11/kdocker/default.nix (hunk 2 above)
Comment (hexclover) [reply pkgs/tools/X11/kdocker/default.nix:47]: But it already says `maintainers = with maintainers; [ hexclover ];` at line 50...?

Comment (drupol) [reply pkgs/tools/X11/kdocker/default.nix:47]: Oops my bad. Sorry !

Could you please sort alphabetically the meta attributes?

Comment (hexclover) [reply pkgs/tools/X11/kdocker/default.nix:47]: Done!

File: pkgs/tools/X11/kdocker/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    rev = "${finalAttrs.version}";
+    hash = "sha256-CTz2M9nv5Rf1amnSpLiIUZLH9Q3te6ZyFNUzSGHdYJc=";
+  };
+
+  nativeBuildInputs = [
+    qmake
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
```
[the remark this answers is not in the document]
Comment (hexclover) [reply pkgs/tools/X11/kdocker/default.nix:29]: Done, also fixed order of lamba parameters.

**Review by drupol — APPROVED** [review]
LGTM, thanks!

**Comment by drupol** [comment]
Thanks and congrats for your first contribution! Keep PRs coming :)

**Pull request merged**